lib/OpenQA/WebAPI/Auth/OpenID.pm

```diff
@@ -105,9 +105,16 @@ sub auth_response {
 
     my $msg = "The openID server doesn't respond";
 
+    my $err_handler = sub {
+        my ($err, $txt) = @_;
+        $self->app->log->error($err, $txt);
+        $self->flash(error => "$err: $txt");
+        return (error => 0);
+    };
+
     $csr->handle_server_response(
         not_openid => sub {
-            die "Not an OpenID message";
+            return $err_handler->("Failed to login", "OpenID provider returned invalid data. Please retry again later");
```

Comment:
Hm, I don't know in detail but something that points the user to the real problem and a suggestion how to continue. I don't think "Please retry again later" will help. But when just waiting worked for you then it's fine to keep it as is for now :-)